### ISSUE TITLE
command failed for 6LowPAN example

### ISSUE THREAD

**Issue by gash9118787** [intrig-unicamp/mininet-wifi#165]
When I ran 6LowPAN example, it showed me error messge.
`* Creating nodes
* Configuring nodes
command failed: Invalid argument (-22)
command failed: Invalid argument (-22)
command failed: Invalid argument (-22)
* Associating Nodes
* Starting network
* Configuring nodes
* Running CLI
* Starting CLI: `
I find it happen because the line in mininet-wifi/mininet/sixLoWPAN/module.py
` os.system('iwpan phy phy%s set netns %s' % (phy, node.pid))`
How can I solve it? Please help me.

**Comment by ramonfontes** [comment]
Did you try "sudo mn -c" and did you install wpan tools with "sudo
util/install.sh -6"?

**Comment by gash9118787** [comment]
Yes, I have down. But the problem still happened.

**Comment by ramonfontes** [comment]
I suggest you to double check it. I couldn't find any problem with 6lowpan.

**Comment by amzarca** [comment]
I have the same problem at the same line. This is:

command failed: Invalid argument (-22)  

I performed some debug:

BEFORE iwpan phy phy45 set netns 12795
command failed: Invalid argument (-22)
BEFORE iwpan phy phy46 set netns 12799
command failed: Invalid argument (-22)
BEFORE iwpan phy phy47 set netns 12803
command failed: Invalid argument (-22)

 I am using the Pre-configured Virtual Machine and I also executed  "sudo mn -c" and "sudo util/install.sh -6".

Thanks in advance.

**Comment by ramonfontes** [comment]
Could you confirm that you have the module fakelb in your kernel? Can you run `sudo modprobe fakelb numlbs=2`?

**Comment by amzarca** [comment]
I have executed:

sudo modprobe fakelb numlbs=2

 It returns nothing.

**Comment by ramonfontes** [comment]
Can you see two interfaces created by fakelb module?

**Comment by amzarca** [comment]
These are the results of the commands:

wifi@wifi-VirtualBox:~$ lsmod | grep fakelb
fakelb                 16384  0
mac802154              77824  1 fakelb


wifi@wifi-VirtualBox:~$ ip link
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: enp0s3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
    link/ether 08:00:27:1a:36:ca brd ff:ff:ff:ff:ff:ff
10: wpan0: <BROADCAST,NOARP> mtu 123 qdisc noop state DOWN mode DEFAULT group default qlen 300
    link/ieee802.15.4 a2:e2:57:6f:78:d1:32:ab brd ff:ff:ff:ff:ff:ff:ff:ff
11: wpan1: <BROADCAST,NOARP> mtu 123 qdisc noop state DOWN mode DEFAULT group default qlen 300
    link/ieee802.15.4 3e:b8:1b:2a:99:a1:8b:0c brd ff:ff:ff:ff:ff:ff:ff:ff

**Comment by ramonfontes** [comment]
Yes, I think so. Try a new git clone and reinstall mininet-wifi with the flag -6. The VM might be out of date for 6lowpan.

**Comment by amzarca** [comment]
I have cloned it in the same virtual machine and I have executed:

sudo util/install.sh -6 (with the same result)
sudo util/install.sh -6Wlnfv (with the same result)
sudo util/install.sh -6a (with the same result)

I have cloned it in another virtual machine (Ubuntu 14.04.5). I cloned the repo and I executed:

apt install dh-autoreconf
sudo  apt-get install libnl-3-dev libnl-genl-3-dev
sudo util/install.sh -6Wlnfv

The same result:
command failed: Invalid argument (-22)

**Comment by ramonfontes** [comment]
Well, tests from Travis says that 6lowpan is working properly. I will take a look on that asap.

**Comment by amzarca** [comment]
I will be waiting for news. I think the error can be related with the set netns parameter of the iwpan.

Thank you in advance.

**Comment by amzarca** [comment]
I have performed these steps in Ubuntu 18.04 and the error does not appears:

apt install dh-autoreconf
sudo apt install libnl-3-dev pkg-config libhwloc5 libhwloc-dev libnl-genl-3-dev
sudo util/install.sh -6Wlnfv

I will test more in deep.

**Comment by ramonfontes** [comment]
Yeah.. really strange. [Test performed by travis on Ubuntu 14.04](https://travis-ci.org/intrig-unicamp/mininet-wifi/jobs/484168563#L3482) doesn't produce such error.

**Comment by capricornus3** [comment]
Hi,

I have the same problem. I use Ubuntu 16.04.1. I did what amzarca wrote, too. but it didn't work.

Not solved yet?

**Comment by ramonfontes** [comment]
Hi,

Which kernel version do you have?

**Comment by capricornus3** [comment]
kernel version:
4.4.0-31-generic

**Comment by ramonfontes** [comment]
Could you try a newer kernel version? Since it works properly with [4.4.0.101](https://travis-ci.org/intrig-unicamp/mininet-wifi/jobs/499270661#L12), I suppose it will work from this kernel version. Many changes were made in fakelb since 2015.

**Comment by capricornus3** [comment]
Yes, it worked :)

I tried it with newer kernel in a different machine and also upgraded the ubuntu to ubuntu 18. Both of them worked.

Thank you

**Comment by ramonfontes** [comment]
Thank you for the feedback! ;)
